flink-connector-bigquery-integration-test/src/main/java/com/google/cloud/flink/bigquery/integration/BigQueryIntegrationTest.java

```diff
@@ -0,0 +1,320 @@
+ * GenericRecord} representing the rows, perform certain operations and then log the total number of
+ * records read. <br>
+ * This module checks the following cases of BigQuery Table read.
+ *
+ * <ol>
+ *   <li>Bounded Jobs: Involve reading a BigQuery Table in the <i> bounded </i> mode.<br>
+ *       The arguments given in this case would be:
+ *       <ul>
+ *         <li>--gcp-project {required; project ID which contains the BigQuery table}
+ *         <li>--bq-dataset {required; name of BigQuery dataset containing the desired table} <br>
+ *         <li>--bq-table {required; name of BigQuery table to read} <br>
+ *         <li>--agg-prop {required; record property to aggregate in Flink job} <br>
+ *         <li>--query {optional; SQL query to fetch data from BigQuery table}
+ *       </ul>
+ *       The sequence of operations in this pipeline is: <i>source > flatMap > keyBy > sum </i> <br>
+ *       A counter counts the total number of records read (the number of records observed by keyBy
+ *       operation) and logs this count at the end. <br>
+ *       Command to run bounded tests on Dataproc Cluster is: <br>
+ *       {@code gcloud dataproc jobs submit flink --id {JOB_ID} --jar= {GCS_JAR_LOCATION}
+ *       --cluster={CLUSTER_NAME} --region={REGION} -- --gcp-project {GCP_PROJECT_ID} --bq-dataset
+ *       {BigQuery Dataset Name} --bq-table {BigQuery Table Name} --agg-prop
+ *       {PROPERTY_TO_AGGREGATE_ON} --query {QUERY} } <br>
+ *   <li>Unbounded Source Job: Involve reading a BigQuery Table in the <i> unbounded </i> mode.<br>
+ *       This test requires some additional arguments besides the ones mentioned in the bounded
+ *       mode.
+ *       <ul>
+ *         <li>--ts-prop {property record for timestamp}
+ *         <li>--partition-discovery-interval {optional; minutes between polling table for new data.
+ *             Used in unbounded/hybrid mode} <br>
+ *         <li>--mode {unbounded in this case}.
+ *       </ul>
+ *       The sequence of operations in this pipeline is the same as bounded one. This job is run
+ *       asynchronously. The test appends newer partitions to check the read correctness. Hence,
+ *       after the job is created new partitions are added. <br>
+ *       Command to run unbounded tests on Dataproc Cluster is: <br>
+ *       {@code gcloud dataproc jobs submit flink --id {JOB_ID} --jar= {GCS_JAR_LOCATION}
+ *       --cluster={CLUSTER_NAME} --region={REGION} --async -- --gcp-project {GCP_PROJECT_ID}
+ *       --bq-dataset {BigQuery Dataset Name} --bq-table {BigQuery Table Name} --agg-prop
+ *       {PROPERTY_TO_AGGREGATE_ON} --mode unbounded --ts-prop {TIMESTAMP_PROPERTY}
+ *       --partition-discovery-interval {PARTITION_DISCOVERY_INTERVAL} }
+ * </ol>
+ */
+public class BigQueryIntegrationTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BigQueryIntegrationTest.class);
+    private static final Integer RECORD_LIMIT = -1;
```

Comment:
No need to set the defaults explicitly. Don't set in the source configs either. Default values will be assigned anyway.

This applies to:
`RECORD_LIMIT `, `ROW_RESTRICTION `, `OLDEST_PARTITION `

Reply:
Fixed.